presto-main/src/main/java/com/facebook/presto/operator/scalar/ListLiteralCast.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.function.OperatorType;
+import com.facebook.presto.spi.function.ScalarOperator;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.type.ArrayType;
+import com.facebook.presto.type.ListLiteralType;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+
+public class ListLiteralCast
+{
+    private ListLiteralCast()
+    {
+    }
+
+    @ScalarOperator(OperatorType.CAST)
+    @SqlType(ListLiteralType.NAME)
+    public static List<Integer> castArrayToListLiteral(@SqlType("array(integer)") Block array)
```

Comment:
I know we've gone back and forth on this a few times in the past, but it just occurred to me that we might not need this mapping for simple array(integer). Getting the values out of the corresponding block on the fly should be as efficient and the conversion from array[1,2,3,...] to block *should* be constant-folded if it's invariant (I need to double-check this).

I need to think further about array(array(integer)).